research/blanks/model.py

```diff
@@ -221,3 +273,48 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
             preblank_embedding_output = shift_right(preblank_embedding_output)
             embedding_output.add_(preblank_embedding_output)
         return embedding_output
+
+
+class BlankPositionalEmbedding(nn.Module):
+    def __init__(
+        self,
+        max_length,
+        embedding_dim,
+        init_type: Literal["kaiming_uniform", "truncated_normal"],
+        init_scale: float,
+        blank_token_id: int,
+    ):
+        super(BlankPositionalEmbedding, self).__init__()
+        self.layer = nn.Embedding(max_length, embedding_dim)
+        default_weight = self.layer.weight.data
+        self.layer.weight.data = get_init_weight(
+            shape=default_weight.shape,
+            fan_in=1,
+            init_type=init_type,
+            scale=init_scale,
+            dtype=default_weight.dtype,
+        )
+        self.blank_token_id = blank_token_id
+
+    def forward(self, x):
+        # stack overflow https://stackoverflow.com/questions/18196811/cumsum-reset-at-nan
+        positions = torch.arange(0, x.shape[-1], device=x.device)
```

Comment:
Let's unsqueeze positions before multiplying, so broadcasting is easier to think about

Reply:
Outdated, now we have loop implementation :(